mesonbuild/interpreter/interpreter.py

```diff
@@ -742,15 +743,21 @@ def run_command_impl(self,
             if not prog.found():
                 raise InterpreterException(f'Program or command {cmd!r} not found or not executable')
             cmd = prog
-        for a in listify(cargs):
+        for a in cargs:
             if isinstance(a, str):
                 expanded_args.append(a)
             elif isinstance(a, mesonlib.File):
                 expanded_args.append(a.absolute_path(srcdir, builddir))
             elif isinstance(a, ExternalProgram):
                 expanded_args.append(a.get_path())
+            elif isinstance(a, compilers.Compiler):
+                FeatureNew('Compiler object as a variadic argument to `run_command`', '0.61.0')
+                prog = ExternalProgram(a.exelist[0], silent=True)
+                if not prog.found():
+                    raise InterpreterException(f'Program {cmd!r} not found or not executable')
```

Comment:
Is this actually possible? Can users get this some way other than meson.get_compiler() which only works for languages that got detected and sanity checked?

Reply:
not sure, it may have just been me programming defensively, but I can't remember exactly why now.